### ISSUE TITLE
Use multibyte-aware string functions.

### ISSUE THREAD

**Issue by dmromanov** [squizlabs/PHP_CodeSniffer#705]
CodeSniffer uses multibyte-anaware functions. E.g.: strlen(), wordwrap(). This may cause errors.

For example, a weird word wrapping in FullReport caused by wordwrap():
```
FILE: /test.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AND 3 WARNING(S) AFFECTING 4 LINE(S)
--------------------------------------------------------------------------------
 148 | ERROR   | Function comment short description must end with a full stop    <-- ANSII Text, OK
 219 | WARNING | Comment refers to a TODO task "сделать проверку                 <-- Unicode, wrapped too early
     |         | на root категорию"
 475 | WARNING | Comment refers to a TODO task "заменить на                      <-- wrapped too early
     |         | реальные ссылки"
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
--------------------------------------------------------------------------------
```

Tested on PHP 5.6.12, PHPCS v1.5.6 & v2.3.4

**Comment by aik099** [comment]
The non-multibyte functions are used for performance reasons. There was a ticket recently about that, but I can't seem to find it.

**Comment by gsherwood** [comment]
> The non-multibyte functions are used for performance reasons.

I think I know why you are saying this, but it's not true. There was only one case where I decided not to use these functions and that was because they were not actually needed. Adding them would have provided the same result, just slower.

**Comment by aik099** [comment]
Ups. Then I must have misunderstood your point in previous discussion about multi-byte function usage. Sorry.

**Comment by gsherwood** [comment]
The full report just uses the built-in [wordwrap](http://php.net/manual/en/function.wordwrap.php) function to do the wrapping, which is aware of word boundaries as well, so I'd have to implement a multi-byte safe version of that to get this working. There are some examples on the PHP docs page in the comments.

**Comment by dmromanov** [comment]
@gsherwood There's been recently the same issue in CakePHP project. You can save some time by using (copying) theirs wordwrap() implementation. Licensed under MIT.
https://github.com/cakephp/cakephp/blob/8b0c23ccd4697cdb6ac507ed32f829ae50458a5e/src/Utility/Text.php#L395

**Comment by b1rdex** [comment]
Max line length sniffs are not unicode-aware too.

**Comment by gsherwood** [comment]
> Max line length sniffs are not unicode-aware too.

Yes they are. You need to make sure you pass the encoding to PHPCS on the command line though.

Something like: `phpcs --encoding=utf-8 ...`

The bit of PHPCS that actually checks for multibyte strings in the core tokenizer. It sets the token column position and length right at the start. Sniffs like the line length checking one use those values to determine if lines are too long. They don't actually measure string lengths themselves.

**Comment by kwn** [comment]
@gsherwood 

I'm afraid you're not right. Could you please have a look at [this](https://github.com/kwn/number-to-words/blob/master/tests/Language/Ukrainian/Transformer/CurrencyTransformerTest.php#L67) example:

I'm running code sniffer using below command:

```
vendor/bin/phpcs -p --encoding=utf-8 --standard=PSR2 src tests
```

and unfortunatelly code sniffer reports a warning:

```
FILE: ...tests/Language/Ukrainian/Transformer/CurrencyTransformerTest.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 67 | WARNING | Line exceeds 120 characters; contains 125 characters
----------------------------------------------------------------------
```

A line itself has 82 columns. It means `phpcs` calculates line length incorrectly. I suspect something has been broken in [this](https://github.com/squizlabs/PHP_CodeSniffer/commit/6829cf72a97374e86a9b9b6d34dbee059942d175) commit. I'll try to have a quick look and to fix the issue.

**Comment by kwn** [comment]
@gsherwood 
Please ignore my above comment. It seems everything is ok.
